feature_details.md

```diff
@@ -0,0 +1,413 @@
+
+* The animations associated with the transition need to be defined completely by the previous Document, which is aware of the site the user is navigating to and has contextual information about the user journey.
+
+* The animations need to be executed such that there are no observable side-effects for the new and previous Documents. For example, setting an obscure animation for a specific element on the incoming Document could be used to exchange information between origins. Enforcing no side-effects is an important consideration, since it can influence the stage in the rendering pipeline where the UA executes these animations.
+
+An important assumption in the design of this feature is that it's reasonable to limit customization options for cross-origin transitions to make it easier for UAs to enforce the constraints above. But the API/implementation should strive to not unnecessarily limit capabilities for same-origin transitions due to cross-origin constraints.
+
+## Glossary
+| Term | Description |
+| ------------- | ------------- |
+| Previous Document  | The Document the user is viewing when a navigation is initated.  |
+| New Document  | The Document which will be current in the session history when a navigation is committed. |
+| Root transition | An animation to transition the content for root element of both Documents. |
+| Exit transition | An animation which specifies how an element in the previous Document exits the screen. |
+| Enter transition | An animation which specifies how an element in the new Document enters the screen. |
+| Snapshot | A pixel copy for a DOM subtree. |
+
+## Sample Code
+This section explains how the feature can be used to add transitions to MPAs. There are a set of use-cases and edge cases to consider :
+
+### Root Transition
+Let's start with a root transition, which adds an animation to the root element of previous and new Documents. Below is a video of the transition we want to create :
+
+[![Video Link for Root Element Transition](https://img.youtube.com/vi/mCvvWMIGItY/0.jpg)](https://www.youtube.com/watch?v=mCvvWMIGItY)
+
+The entry point for this API is a new `document.documentTransition` object. The previous Document provides a list of elements which will be animated during the transition. Since this is root transition, this list only includes the `html` element. The result of calling `documentTransition.setSameOriginTransitionElements` as illustrated below is that the browser caches a copy of the pixels (snapshot) for the `html` element when a navigation to `foo.com/a.html` is initiated.
+
+```
+// The list of elements is keyed on url. When a user navigates to
+// a url, the corresponding entry in this dictionary is used to
+// decide which elements should be cached.
+document.documentTransition.setSameOriginTransitionElements(
+  {
+   “foo.com/a.html” : [{element : document.html}],
```

Comment:
I don't quite understand why you need to specify the url, including the domain, for a same origin transition. Also, I missed that this was for MPA and was even more confused :P Maybe mentioning again somewhere that this is for MPA specifically would be nice.

Reply:
This needs to be refined more, I don't expect the eventual API would require exact URL strings. Especially because a common scenario would be to use regex pattern to specify the same set of elements for a set of pages. I'll add a comment clarifying that this needs more work. We can probably take inspiration from pre-render speculation rules for structuring this : https://github.com/jeremyroman/alternate-loading-modes/blob/main/triggers.md#speculation-rules

But the goal of the API is to allow the author to configure sharing a different list of elements for any same origin navigation.

Reply:
Added a comment to clarify this in the API proposal section.